### ISSUE TITLE
Using v3.2 results in "fatal: could not read Username for 'https://github.com': No such device or address"

### ISSUE THREAD

**Issue by sprak3000** [aormsby/Fork-Sync-With-Upstream-action#45]
When I attempt to use v3.2 of the action, I receive this as an error:

```
Pushing synced data to target branch.
fatal: could not read Username for 'https://github.com': No such device or address
```

Looking through past issues, I found #32 and saw the fix applied made it into v3.0 with the issue creator using v3.0 in their repo currently. Switched to v3.0 and no longer see the above error message. That said, v3.0 gives me this error:

```
fatal: refusing to merge unrelated histories
```

This error seems related to #37 where the fix was applied to v3.2 of the action. I'm stuck between two errors currently.

If it helps, here is the [configuration](https://github.com/HqOapp/migrate/blob/master/.github/workflows/upstream-sync.yml) for the action I am using.

Let me know if there are any other details I can provide.

Thank you.

**Comment by ZhaoUncle** [comment]
same issues

**Comment by Kukkerem** [comment]
same

**Comment by aormsby** [comment]
If privacy is not an issue, could someone please link their workflow logs?

**Comment by aormsby** [comment]
@sprak3000
**For v3.0 failure**
- Your checkout depth is 1 by default when using github checkout action.
- Try to add `fetch-depth: 0` to the checkout step to make sure you have a full history for merging. (You can probably remove this after the first successful run.)
- Additionally, you can add `upstream_pull_args: '--allow-unrelated-histories'` to the sync step and try that.

Not sure on v3.2, haven't had a chance to sit with it.

**Comment by sprak3000** [comment]
@aormsby 

Looks like those commands might have done the trick. The most recent scheduled attempt passed, but there were no updates from upstream. I'm going to leave this open a bit longer to see what happens when there are updates, but the initial outlook is promising.

**Comment by rmarqusi** [comment]
I was facing the same error here(private repo), I used you suggest, and now it's fine. Thanks @aormsby 
Lines included:  
fetch-depth: 0
upstream_pull_args: '--allow-unrelated-histories'
the message error still but now the action could identify new commits/ No new commits
![image](https://user-images.githubusercontent.com/47756058/158905468-fa88f19c-1a10-4c1b-9a1a-de0181207e92.png)

**Comment by aormsby** [comment]
Cool, thanks for the updates 👍

**Comment by jantari** [comment]
Ran into this same issue on 3.2 and it appears that the fix that made it into 3.0 was removed in 3.2 again.

This was the fix in 3.0: https://github.com/aormsby/Fork-Sync-With-Upstream-action/commit/aa1ee5d4f968eff69802e9a63faebcac1734e273?diff=split

but if we look at the current `push_updates.sh` file in main or 3.2 then that change is undone: https://github.com/aormsby/Fork-Sync-With-Upstream-action/blob/v3.2/run/push_updates.sh

I didn't comb through the whole history to find why that commit was undone, but it seems like the fix for 3.2 would be to just re-apply this same old fix aa1ee5d4f968eff69802e9a63faebcac1734e273 ?

**Comment by aormsby** [comment]
Hmm, I really don't remember. 😐
